source/dhtslib/sam.d

```diff
@@ -164,12 +202,12 @@ class SAMRecord
 
     /// Return pointer to sequence Cstring
     /// see samtools/sam_view.c: get_read
-    @property char* sequence()
+    @property string sequence()
```

Comment:
this is a breaking change; was this consumed anywhere already?

also, if you change teh fn signature to return const(char)[] you may be able to return s directly without the idup which will speed up repetitive sequence fetching quite abit. if some other API component needed a string sequence, could still pass the const(char)[] with autoatic castic

Reply:
In all my code I have been accessing the nibble encoded string directly. So it will likely not break any of my programs, but I cannot speak for other programs. I have not tested the dhtslib test programs.

Reply:
If we want to reassign the same string sequence e.g.
```
auto seq = read.sequence; //read is a SAMRecord
read.sequence=seq[0..10];
```
I have to duplicate the sequence somewhere. Otherwise I get a complaint from the ldc about overlapping array copy somewhere around here in the ```void sequence(string seq)``` function:
https://github.com/blachlylab/dhtslib/blob/67563cd67ae799e30034d9c9f391762913c2a598/source/dhtslib/sam.d#L253-L258

For cigar I have this problem and I have to ```.dup``` the cigar array since the struct returned by ```read.cigar``` is really a slice to the cigar in the b.data array.

Reply:
Actually I just realized my previous statement isn't true as I have to re nibble encode the sequence again which is a copy. I will address this and test.